mlos_bench/mlos_bench/environments/local/local_env.py

```diff
@@ -63,13 +70,36 @@ def __init__(self,
         assert self._service is not None and isinstance(self._service, SupportsLocalExec), \
             "LocalEnv requires a service that supports local execution"
         self._local_exec_service: SupportsLocalExec = self._service
+        self._temp_dir: Optional[str] = None
+        self._temp_dir_context: Union[TemporaryDirectory, nullcontext, None] = None
 
-        self._temp_dir = self.config.get("temp_dir")
+        temp_dir = self.config.get("temp_dir")
+        if temp_dir is not None and not os.path.isdir(temp_dir):
+            _LOG.error("temp_dir path is invalid: %s", temp_dir)
+            raise ValueError(f"Temp directory path is invalid: {temp_dir}")
```

Comment:
This is probably a little too strict.  Creating the base temp_dir on demand should be ok, as well (as long as we have permissions for it).

Reply:
I see. Let me roll back the change and I will do it properly with unit tests and stuff later